src/python/ensembl/io/genomio/gff3/extract_annotation.py

```diff
@@ -149,30 +157,31 @@ def add_feature(self, feature: SeqFeature, feat_type: str, parent_id: Optional[s
             else:
                 raise AnnotationError(f"No parent possible for {feat_type} {feature.id}")
 
-    def _generic_feature(self, feature: SeqFeature, feat_type: str) -> Dict[str, Any]:
+    def _generic_feature(
+        self, feature: SeqFeature, feat_type: str, parent_ids: Optional[List[str]] = None
+    ) -> Dict[str, Any]:
         """Create a feature object following the specifications.
 
         Args:
             feature: The SeqFeature to add to the list.
             feat_type: Feature type of the feature to store (e.g. gene, transcript, translation).
 
         """
+        if parent_ids is None:
+            parent_ids = []
+
         feature_object: Annotation = {"object_type": feat_type, "id": feature.id}
 
         # Description?
-        if "product" in feature.qualifiers:
-            description = feature.qualifiers["product"][0]
-            if self.product_is_informative(description):
-                feature_object["description"] = description
-
-        if "Name" in feature.qualifiers and "description" not in feature_object:
-            feature_object["description"] = feature.qualifiers["Name"][0]
-
-        # Don't keep useless description
-        if ("description" in feature_object) and not self.product_is_informative(
-            feature_object["description"], feature.id
-        ):
-            del feature_object["description"]
+        for qname in ("description", "product"):
+            if qname in feature.qualifiers:
+                description = feature.qualifiers[qname][0]
+                all_ids = list(parent_ids)
+                all_ids.append(feature.id)
+                if self.product_is_informative(description, feat_ids=all_ids):
```

Comment:
```suggestion
                if self.product_is_informative(description, feat_ids=parents_ids + [feature.id]):
```
This has the same behaviour, but it might be more readable (unless someone knows that `list()` generates a copy of the list, they may not understand why you call `list` again over a variable that clearly contains a list).